llvm/lib/Target/ARM/MCTargetDesc/ARMAsmBackend.cpp

```diff
@@ -446,6 +447,16 @@ unsigned ARMAsmBackend::adjustFixupValue(const MCAssembler &Asm,
                                          const MCSubtargetInfo* STI) const {
   unsigned Kind = Fixup.getKind();
 
+  // For MOVW/MOVT Instructions, the Fixup Value needs to be 16 bit aligned.
+  // If this is not the case, we should reject compilation.
+  if((Kind == ARM::fixup_arm_movw_lo16 || Kind == ARM::fixup_arm_movt_hi16 ||
+      Kind == ARM::fixup_t2_movw_lo16 || Kind == ARM::fixup_t2_movt_hi16) &&
+      (!(minIntN(16) <= static_cast<int64_t>(Value) &&
+    static_cast<int64_t>(Value) <= maxIntN(16)))) {
```

Comment:
This condition could do with re-ordering. Put the fixup value first each time and remove the need for `!`. This if is "if something produce an error", so write it in terms of when you will error not when you won't.

```
&& (static_cast<int64_t>(Value) < minIntN(16) || static_cast<int64_t>(Value) > maxIntN(16))
```

Reply:
I long for the day. I have updated with your suggestion so it reads better.